### DISCUSSION TITLE
how to configure multiple useragents?

### DISCUSSION THREAD

**Discussion by Tumtum2814** [jef/streetmerchant#1228] [Help]
hey all, I was hoping someone could help me with useragent formatting. I know it won't fully protect me and i've increased my sleep between scrapes but i wanted to put a couple useragents in the .env 
running streetmerchant in docker it's been working great but getting this error when adding new UAs
variable 'Mozilla/5.0 (Linux; Android 8.0.0; SM-G960F Build/R16NW) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/62.0.3202.84 Mobile Safari/537.36 \n' contains whitespaces.
See 'docker run --help'.

my env file line is 
USER-AGENT="Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:83.0) Gecko/20100101 Firefox/83.0 \n
Mozilla/5.0 (Linux; Android 8.0.0; SM-G960F Build/R16NW) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/62.0.3202.84 Mobile Safari/537.36 \n
Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/70.0.3538.110 Safari/537.36"

**Comment by jef** [comment]
I have this changing in 4.0.0, but in 3.x, it needs changing. The way it works now isn't great. Your format looks correct, but I'll post again when this changes.

Thanks!

**Comment by coredumperror** [comment]
Has it changed? I'm getting a bunch of capcha and 403 errors, and I'm told that setting up multiple user agents is the way to fix that. I'd love to know how to configure them.

**Reply by jef** [reply]
I forgot to update this discussion, but there are multiple user agents generated now and shouldn't be necessary. but if you want to use multiple, you would do it like this:

```
USER_AGENT="Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.88 Safari/537.36\nMozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.88 Safari/537.36\nMozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.88 Safari/537.36"

**Reply by coredumperror** [reply]
Oh, hm. So I guess multiple user agents doesn't help against these 403, 503, and CAPCHA errors. What _does_ help against those?

**Reply by jef** [reply]
It should help with CAPTCHA usually, but for the 403/503 perhaps not. I think one think one thing that works for certain is [rotating proxies](https://jef.codes/streetmerchant/reference/proxy/).

**Reply by MoncefAfif** [reply]
> 
> 
> It should help with CAPTCHA usually, but for the 403/503 perhaps not. I think one think one thing that works for certain is [rotating proxies](https://jef.codes/streetmerchant/reference/proxy/).

the URL is not working and i'm new to all of this can you refer me to another website where they explain how to use proxies for streetmerchant and thanks

**Reply by Mr-Tech-13** [reply]
Use freeproxylists to find proxies. To use the proxy server in dotenv paste the address and port in the respective places.
